Make libopencv python-independent; collapse CI jobs

h-vetinari opened the pull request:
Alright, this is what I promised to look in to in #363. I could see that the default python standalone build will [fail](https://github.com/opencv/opencv/blob/4.7.0/modules/python/standalone.cmake#L4), and I patched that, but it'll probably take some iteration to get right.

The commits are intended to be reviewed in sequence, the overall diff is almost unreadable.

Closes #364

h-vetinari committed:
ensure there are no collisions between output names and global package name

h-vetinari committed:
simplify opencv_libs

h-vetinari committed:
update ancient python versions in selectors

h-vetinari committed:
move tests to proper output

h-vetinari committed:
improve testing of library artefacts

h-vetinari committed:
clean up CMake integration test

h-vetinari committed:
ensure we point to the python-config from the global build step

h-vetinari committed:
make libopencv independent of python; build only once

h-vetinari committed:
move libgomp to host deps

hmaarrfk commented on recipe/meta.yaml:
```diff
@@ -81,128 +75,155 @@ requirements:
+        ] + (not win) * [
+            "bioinspired",
+            "hdf"
+        ]%}
```
the old style i remember was used because the bot was having trouble with certain aspects of templating.

I think this recipe is always problematic for the auto-tick bot, but I just watned to bring this up as a potential reason why we might want to revert this particular change.

h-vetinari replied:
I've had no problems with multiline jinja variables in places like [arrow](https://github.com/conda-forge/arrow-cpp-feedstock/blob/main/recipe/meta.yaml#L158-L171), and the auto-tickbot still manages to open new PRs: https://github.com/conda-forge/arrow-cpp-feedstock/pull/1087

I don't doubt that the problems were there, but 38d050881c3be2f91f3550437be320d81109938c is 2.5 years old. Perhaps things changed in the meantime?

It's not a crucial aspect of this PR, but I find the current wall of text to be noisy bordering on unreadable, so I'd like to try getting rid of it.

hmaarrfk replied:
just giving a historical perspective when i can. Excited to see references to PYTHON2 go away!

hmaarrfk commented:
curious to see what happens. Thanks!

h-vetinari committed:
Use OPENCV_PYTHON_STANDALONE_INSTALL_PATH

opencv_python_config.cmake from global build step seems to
overwrite our OPENCV_PYTHON_INSTALL_PATH with an empty string

h-vetinari committed:
try with OPENCV_INITIAL_PASS=ON

h-vetinari committed:
try to include missing module

h-vetinari committed:
glog/gflags

h-vetinari committed:
on the way to no-lah

h-vetinari committed:
clean up build cache

h-vetinari committed:
more debug

h-vetinari committed:
MNT: Re-rendered with conda-build 3.25.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.06.30.10.20.46

h-vetinari commented:
I don't understand why including some CMake modules would cause ninja not to be found, as it's clearly in the environment...
```
CMake Error:
  The detected version of Ninja () is less than the version of Ninja required
  by CMake (1.3).
```

traversaro commented:
> I don't understand why including some CMake modules would cause ninja not to be found, as it's clearly in the environment...
> 
> ```
> CMake Error:
>   The detected version of Ninja () is less than the version of Ninja required
>   by CMake (1.3).
> ```

I am no sure this is what is causing that error, but could this be related to the fact that if `OPENCV_INITIAL_PASS` is enabled, there is no call to `cmake_minimum_required` and `project`, that are required by any CMake project as first step?
